systems/trajectory_optimization/dircon_position_data.cc

```diff
@@ -18,13 +20,21 @@ using drake::MatrixX;
 
 template <typename T>
 DirconPositionData<T>::DirconPositionData(const MultibodyPlant<T>& plant,
-    const Body<T>& body, Vector3d pt, bool isXZ) :
-    DirconKinematicData<T>(plant, isXZ ? 2 : 3),
-    body_(body),
-    pt_(pt),
-    isXZ_(isXZ) {
+    const Body<T>& body, Vector3d pt, bool isXZ,
+    Vector3d surface_normal) :
+        DirconKinematicData<T>(plant, isXZ ? 2 : 3),
+        body_(body),
+        pt_(pt),
+        isXZ_(isXZ) {
   TXZ_ << 1, 0, 0,
           0, 0, 1;
+
+  Vector3d z_hat(0,0,1);
+  surface_normal.normalize();
```

Comment:
Does it make sense to supply a `surface_normal` value if `isXZ = true`? If these two arguments cannot be supplied together, then this should either be prevented via the API or, at minimum, checked at runtime.

Reply:
Yes, I use both of them in the example of planar five-link robot example.